### PR TITLE
Rework the Flare Gun & add a Security Shell Gun.

### PR DESCRIPTION
**Pull request by BramvanZijp**
## About the PR
Several changes were made in this PR:
1. Rework flare guns, they now work sort of like a pistol except without the magazine. They start closed with a shell loaded in them, then you can fire it. The empty shell does not eject. Then you press E on it to unbolt and open it, which drops the current shell on the ground and allow you to put a new shell inside, then you can close it again to fire again.
2. Add a security shell gun, which is a resprited flare gun restricted to security, and is craftable in the sec techfab. This is designed to launch beanbag or tranq rounds, though it can also make use of lethal shells in a pinch.
3. Flare guns are now twice as likely to spawn in emergency lockers.

## Why / Balance
A sprite for an opened flare gun has existed in the code, but it is never used. It also just makes sense to have to open it for the old shell to fall out and you to be able to put the new shell back in, this also prevents spam reloading and firing it with the extra added step.

As for the security shell gun, it is craftable in the secfab and designed to be used with beanbag and tranquilizer shells. From my experience these shells, especially tranquilizer, very rarely see use due to the hastle of carrying around the entire shotgun just to use these non-lethal shells and the fact that it would be frowned upon outside alert levels that warrant lethals, at which point you are better off taking lethal shells instead. This would allow security to carry one of these around on lower alert levels and still use the shells.

## Technical details
The flare gun now uses the code used by pistols (excluding a magazine) instead of that used by shotguns. Also, improvised, beanbag, tranquilizer and flare rounds now have the ShotgunShellLight, if somebody intends to re-introduce #22349 properly, this can be used for that. (The Security Shell Gun should still be able to fire lethals.)

## Media
A video of the regular flare gun with the new bolt mechanics:
https://github.com/user-attachments/assets/00f46007-03b7-4c97-b6ab-a08704112558

The new security shell gun sprites: (Just recolored to be red)
base bolt-open

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl: BramvanZijp
- tweak: Flare guns have been reworked to no longer automatically eject shells after firing, and must now be opened manually to insert or eject shells. A safety feature has also been added to prevent loading lethal shells into them.
- tweak: Flare guns are twice as likely to appear in emergency closets, with the same rarity as MRE's.
- add: Added the security shell gun, a modified flare gun which is capable of handling both nonlethal and lethal shotgun shells. It is able to be crafted in the security techfab without needing any research.

### CODE REVIEW

**Comment by BramvanZijp** [comment]
For the record, later on it may be good to have a system where the plastic flaregun can backfire if using lethal shells, but for now it should probably be disabled which is done in this PR.

**Comment by superjj18** [comment]
Pretty sure irl flare guns are 12 guage so yes they can totally fire lethal rounds.

I'd rather the lethal rounds work while also breaking the flare gun(and maybe causing some minor burn damage to the shooter).  If you're going for realism that's probably better than arbitrarily limiting the type of round for no reason.  If they want a quick burst of damage, they get one shot and a broken flaregun 

Whitelists are lame and aren't communicated to the player 

(Also I feel the manual reloading makes it more annoying to use, but at the same time I get why so that's easier to accept)

**Comment by BramvanZijp** [comment]
> Pretty sure irl flare guns are 12 guage so yes they can totally fire lethal rounds.
> 
> I'd rather the lethal rounds work while also blowing up the flare gun, as if your going for realism that's probably better than arbitrarily limiting the type of round for no reason

Ill figure out a solution for that then, few questions though.
1. Should the flare gun

> Pretty sure irl flare guns are 12 guage so yes they can totally fire lethal rounds.
> 
> I'd rather the lethal rounds work while also blowing up the flare gun, as if you're going for realism that's probably better than arbitrarily limiting the type of round for no reason.
> 
> Whitelists are lame and aren't communicated to the player

Ill find a way to have them blow up instead, should it just be the clown/monkey blow-up or an actual explosion?

Also, this is literally added to the item description but fair ig

**Comment by KaiserMaus** [comment]
What about makeshift shotgun shell? Flare can fire this munitions?

**Comment by superjj18** [comment]
> > Pretty sure irl flare guns are 12 guage so yes they can totally fire lethal rounds.
> 
> > 
> 
> > I'd rather the lethal rounds work while also blowing up the flare gun, as if you're going for realism that's probably better than arbitrarily limiting the type of round for no reason.
> 
> > 
> 
> > Whitelists are lame and aren't communicated to the player
> 
> 
> 
> Ill find a way to have them blow up instead, should it just be the clown/monkey blow-up or an actual explosion?
> 
> 
> 
> Also, this is literally added to the item description but fair ig

I think it would slightly burn their hand.  Wouldn't severely wound them or even knock them over, but the flare gun definitely wouldn't be able to fire again.(the handle and the barrel would likely break apart)

You could probably just separate the handle and barrel into two sprites, have the flaregun turn into a handle and spawn the barrel item on the ground.  Something like that

**Comment by BramvanZijp** [comment]
Its going to be a pain to properly sort the flaregun blowing up with lethal shells in this PR, and it also feels a bit too far out of its scope. So It has been removed, if somebody else wants to re-add #22349 properly, please do so (And please make sure the sec shell gun is not effected.)

**Comment by BramvanZijp** [comment]
(To clarify, this means that for now, flare guns can still fire lethal shells.)

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml
Comment (chromiumboy) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:23]: I'd remove the standard shotgun shells, at the very least until the risk of backfiring can be added

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml
Comment (chromiumboy) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:54]: If it's suppose to be a non-lethal device, I'm not sure if it firing lethal shells is a good idea. What purpose is this meant to serve?

All new weapons need approval by either sloth or DrSmugleaf (https://github.com/space-wizards/space-station-14/issues/8524)

File: Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun_security.rsi/meta.json
Comment (chromiumboy) [inline Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun_security.rsi/meta.json:4]: Eris textures are usually not compatible with ours, this will need art director approval

**Review by chromiumboy — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml
Comment (superjj18) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml]: Well... it's realistic as many irl flare guns are 12 gauge, and it's kind of a cool cheap improvised weapon.

Comment (BramvanZijp) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:54]: [Quoted from #design-discussion.](https://discord.com/channels/310555209753690112/1008709214006427689/1295768242975146015) 

Regular passengers will only have access to the regular flare gun (made out of plastic). You can fire shells in it just fine, but if you try to fire any shell other than a flare shell, the flaregun explodes and turns into a broken one, and we can maybe roll a 50% for the gun blowing up in their hand (clumsy system) or it “””working””” and shooting the shot, but one time.

Security will get a metal flaregun that just works with all shells. This is how they work in real life. 


Furthermore, the purpose of being able to fire lethal shells is both realism and for in an emergency self defense situation. I would assume giving out lethal sec shell guns is still considered a lethal weapon and thus code red only.
As for the weapon approval, [a maintainer said it should be fine unless somebody complains](https://discord.com/channels/310555209753690112/770682801607278632/1295803846450479155), so let me know if I need to still get review keeping in mind this is just a reskin of an already existing weapon while that weapon is being changed to no longer be lethal.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml
Comment (BramvanZijp) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:23]: That I can do.

File: Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun_security.rsi/meta.json
Comment (BramvanZijp) [reply Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun_security.rsi/meta.json:4]: This is the exact texture used by the current ingame flaregun, just with the orange pixels changed out for red ones, let me know if that is still a problem.

Comment (chromiumboy) [reply Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun_security.rsi/meta.json:4]: The sprite has been OK'ed by [one of the art leads](https://discord.com/channels/310555209753690112/346743671787356160/1299902662384554044)

File: Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun_security.rsi/meta.json
Comment (metalgearsloth) [inline Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun_security.rsi/meta.json:28]:
```suggestion
        {
```

**Comment by Everturning** [comment]
i dont understand why security needs a shell gun

**Comment by BramvanZijp** [comment]
> i dont understand why security needs a shell gun

Its to give security more options with less lethal ammo like beanbags or tranquilizers. They are currently very rare to see since if you are allowed to carry a full blown shotgun you would just use lethals with it, so this serves as a way to make these ammo types available without the big red alert grade weapons.

**Comment by BramvanZijp** [comment]
(Also if somebody rebalances tasers as a t3 tech for security, a taser shell (like an XREP) would also be really cool to use with this.)

**Comment by Everturning** [comment]
> > i dont understand why security needs a shell gun
> 
> Its to give security more options with less lethal ammo like beanbags or tranquilizers. They are currently very rare to see since if you are allowed to carry a full blown shotgun you would just use lethals with it, so this serves as a way to make these ammo types available without the big red alert grade weapons.

well I dont see why. this isn't really a niche that needs to be filled. disablers already do the job of taking down enemies at a distance far better than a beanbag shotgun ever could, and tranqs are used mostly in tandem with lethals. security really does not need a miniature shotgun in their pocket even if it's able to be used on green or whatever. besides, you said in the PR description  that it can shoot lethal shells, so that goes against that whole point.

**Comment by BramvanZijp** [comment]
> > > i dont understand why security needs a shell gun
> > 
> > 
> > Its to give security more options with less lethal ammo like beanbags or tranquilizers. They are currently very rare to see since if you are allowed to carry a full blown shotgun you would just use lethals with it, so this serves as a way to make these ammo types available without the big red alert grade weapons.
> 
> well I dont see why. this isn't really a niche that needs to be filled. disablers already do the job of taking down enemies at a distance far better than a beanbag shotgun ever could, and tranqs are used mostly in tandem with lethals. security really does not need a miniature shotgun in their pocket even if it's able to be used on green or whatever. besides, you said in the PR description that it can shoot lethal shells, so that goes against that whole point.

Its able to shoot lethals in an emergency. And a niche that doesnt NEED to be filled doesnt mean it wouldnt be a nice addition to spice up gameplay. Disabler SMG wasnt a niche that NEEDED to be filled for the same reason, but its still a nice addition.

**Comment by Everturning** [comment]
> > > > i dont understand why security needs a shell gun
> > > 
> > > 
> > > Its to give security more options with less lethal ammo like beanbags or tranquilizers. They are currently very rare to see since if you are allowed to carry a full blown shotgun you would just use lethals with it, so this serves as a way to make these ammo types available without the big red alert grade weapons.
> > 
> > 
> > well I dont see why. this isn't really a niche that needs to be filled. disablers already do the job of taking down enemies at a distance far better than a beanbag shotgun ever could, and tranqs are used mostly in tandem with lethals. security really does not need a miniature shotgun in their pocket even if it's able to be used on green or whatever. besides, you said in the PR description that it can shoot lethal shells, so that goes against that whole point.
> 
> Its able to shoot lethals in an emergency. And a niche that doesnt NEED to be filled doesnt mean it wouldnt be a nice addition to spice up gameplay. Disabler SMG wasnt a niche that NEEDED to be filled for the same reason, but its still a nice addition.

okay? define an emergency in in-game mechanics. how will the GUN decide what's an emergency or not?

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml
Comment (chromiumboy) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:52]: The shell gun should be inheriting from WeaponFlareGun. There are a lot of duplicate fields between the two prototypes

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml
Comment (chromiumboy) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:10]: Avoid renaming textures, it can break things downstream

File: Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun.rsi/meta.json
Comment (chromiumboy) [inline Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun.rsi/meta.json:11]: See comment above

**Review by chromiumboy — CHANGES_REQUESTED** [review]
I've review the PR and there's a few things that need to be adjusted. But overall I like the changes to the flare gun, they are quite fitting thematically

I've left some code comments on the shell gun, however, I honestly don't really see how it fits into security. Security has access to disablers, which are just overall better, and has access to shotguns that can fire all the shells that the shell gun does. It has very low DPS, so security are going use their pistols if lethal action is necessary. And sure, you can print shell guns for the crew in an emergency, but if the situtauion is that dire that the crew needs to be armed, security can hand out better weapons, print lasers with just a little research, or just have cargo order more weapons

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml
Comment (BramvanZijp) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:10]: This is required for the system that changes the sprite depending if the "bolt" is open or closed.

File: Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun.rsi/meta.json
Comment (BramvanZijp) [reply Resources/Textures/Objects/Weapons/Guns/Shotguns/flaregun.rsi/meta.json:11]: See reply above

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml
Comment (BramvanZijp) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:10]: I can make it so theres both a base AND an icon texture, but we need one called base for said system.

Comment (chromiumboy) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:10]: OK, I'll look into what should be done here

Comment (chromiumboy) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml:10]: [It looks like it should be fine to rename](https://discord.com/channels/310555209753690112/900426319433728030/1301401658072367115)

**Review by chromiumboy — APPROVED** [review]
While I still have my reservations about the role of the shell gun, I don't see the harm in it, and Sloth has OK it. So if you can find another maintainer to approve it, it looks good to go

**Review by SlamBamActionman — CHANGES_REQUESTED** [review]
This PR seems to have an issue that I haven't seen brought up; since this copies gun code, it has a "Rack" action that doesn't make sense. Not only is it not necessary to use the gun ingame, but it also takes up the Use action verb, making it impossible to close/open the flaregun using the keyboard. So that would need to be removed, and is the primary blocker for this PR right now.

As-is, this gun completely nullifies the Improvised Shotgun. I've seen that the intention is to make it explode on other shells, so that should *really* be prioritized. 

> Added the security shell gun to the security techfab, a modified flare gun designed to let security fire nonlethal .50 shells, though it can also fire lethal shells in a pinch.

I would personally not use this primarily for non-lethals. It's a decent one-shot burst that you can pull out from your pocket in an instant, very similar to the sawn-off, and as previously mentioned if Security wants to disable someone they have other options. I do think it makes for a good addition to the gun arsenal though. I would just state that it's rated for all shotgun shells in its description/in the changelog, instead to specify it's made for non-lethals.

**Comment by BramvanZijp** [comment]
> This PR seems to have an issue that I haven't seen brought up; since this copies gun code, it has a "Rack" action that doesn't make sense. Not only is it not necessary to use the gun ingame, but it also takes up the Use action verb, making it impossible to close/open the flaregun using the keyboard. So that would need to be removed, and is the primary blocker for this PR right now.
> 
> As-is, this gun completely nullifies the Improvised Shotgun. I've seen that the intention is to make it explode on other shells, so that should _really_ be prioritized.
> 
> > Added the security shell gun to the security techfab, a modified flare gun designed to let security fire nonlethal .50 shells, though it can also fire lethal shells in a pinch.
> 
> I would personally not use this primarily for non-lethals. It's a decent one-shot burst that you can pull out from your pocket in an instant, very similar to the sawn-off, and as previously mentioned if Security wants to disable someone they have other options. I do think it makes for a good addition to the gun arsenal though. I would just state that it's rated for all shotgun shells in its description/in the changelog, instead to specify it's made for non-lethals.

Ill remove the rack verb from it.
The flare gun explode on other shell will be implemented once #31147 is merged (please merge this), keep in mind that right now the vanilla flaregun cannot fire lethals at all when this PR is merged.
Ill update the description & changelog with this in mind.

**Comment by BramvanZijp** [comment]
> If there's a break action mechanic, it should apply to the double barrel shotgun too

Much harder to do since it should hold 2 rounds, but should prolly be done yeah.

**Pull request merged**
